### PR TITLE
Adds some very basic random testing and unit tests

### PR DESCRIPTION
**Pull request by danwt**
Adds some unit tests and random testing. It's a bit unfinished, the slash query isn't properly checked yet and the code is unpolished. But it's something.

Tests:

- assignment specific cases
- random assignents interleaved with applying mapping to random updates
- pruning property
- validator set replication

Does not test:

- Reverse lookup for slashing

### CODE REVIEW

**Comment by danwt** [comment]
Note, the test is failing but I do not know whether it is due to a bug in the SUT or in the test. I don't have time to find out this evening.

File: x/ccv/provider/keeper/key_assignment.go [hunk 1]
```diff
@@ -506,6 +509,8 @@ func (k Keeper) ApplyKeyAssignmentToValUpdates(
 			// of a newly registered consumer chain.
 			consumerKey, found := k.GetValidatorConsumerPubKey(ctx, chainID, providerAddr)
 			if found {
+				fmt.Println("expect 1")
```
Comment (mpoke) [inline x/ccv/provider/keeper/key_assignment.go:512]: Remove.

File: x/ccv/provider/keeper/key_assignment.go [hunk 2]
```diff
@@ -423,12 +425,13 @@ func (k Keeper) AssignConsumerKey(
 		}
 
 		// check whether the validator is valid, i.e., its power is positive
-		providerValidatorAddr, err := sdk.ValAddressFromBech32(validator.OperatorAddress)
-		if err != nil {
-			return err
-		}
-		if power := k.stakingKeeper.GetLastValidatorPower(ctx, providerValidatorAddr); power > 0 {
+		// providerValidatorAddr, err := sdk.ValAddressFromBech32(validator.OperatorAddress)
```
Comment (mpoke) [inline x/ccv/provider/keeper/key_assignment.go:428]: remove

**Comment by jtremback** [comment]
LGTM!

**Pull request merged**